draft.md

```diff
@@ -0,0 +1,81 @@
+# Draft Ipld Unixfs Spec
+
+## Basic Structure
+
+   - Some sort of header that indicates that this a directory and included a version number.  The header could also have fields to give additional information on the meaning of the extended attributes.
+
+  - CBOR Map
+    - Key: CBOR Byte or Text String: File Name
+    - Value: CBOR Array of:
+      - Type: CBOR Unsigned Int
+      - Link or Data: CBOR Type varies
+      - Optional file size: CBOR Unsigned Int
+      - Optional Standard Attributes: CBOR Map
+      - Optional Extended Attributes: CBOR Map
+
+The file size is only defined for regular files and is the size of the file contents.
+
+All maps should be ordered based on the binary values of the key,
+duplicates are not allowed.
+
+### Notes
+
+ * An array makes sense to be as this is more compact and the value of
+   the fields are unambiguous, it also allows for a separation of
+   standard and extended attributes
+
+ * The key type can either be a byte or text string as POSIX makes no
```

Comment:
> can either be a byte or text string

Can this bytes/text distinction roundtrip to JSON and back? RFC7049 is unclear on this...

Reply:
The byte case will be reserved for the case when the filename is not a valid UTF-8 string in order to faithfully represent any valid file in the POSIX setting.  I image it will need to be encoded using BASE-64 or something similar when encoded in JSON.

An alternative would be to just use UTF-8 text and then encode the rare case of a non-unicode bytes in a special fashion, it is unclear how this can be done.

Reply:
I think it being either String or bytestring (as CBOR calls it) will create more problems than we expect. It is still problematic with JSON (as we can't well represent bytearrays in JSON, we are thinking about something that would workaround this problem by allowing you to ask for many formats with order of preference and it would return first format that is about to fully represent the data).

It would mean that there is no minimal schema for this format. Depending on what we decide to do with filenames (UTF-8 only, arbitrary, safe characters only) we should make it either a string or bytestring.

Reply:
Note: I created a separate issue to discuss this, #3.